Courses/AU/r-studio-demo-course__0000/2024-09-01/Dockerfile

```
@@ -0,0 +1,21 @@
+FROM dreg.cloud.sdu.dk/ucloud-apps/rstudio:4.4.1
+
+LABEL course="RStudio Demo Course" \
+      author="Jes Elgin Jensen, Rytis Mikutavicius, Mine Çetinkaya-Rundel" \
+      description="Example RStudio-centered UCloud course based on based on Duke University's Statistical Science Introductory Data Science courses"
+
+## Use bash instead of sh
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+WORKDIR /home/${USERID}
+
```

Install the required R packages directly from the Dockerfile like so. This removes the need for an R environment:
```suggestion
RUN sudo apt-get update -qq \
 && install2.r --error --deps TRUE \
    ggplot2 \
    dplyr \
 && rm -rf /tmp/downloaded_packages/ /tmp/*.rds \
 && sudo apt-get clean \
 && sudo rm -rf /var/lib/apt/lists/*
```
In fact, this step is not necessary since the entire `tidyverse` package suite is already installed into the base image. But I think it still could make sense to include the code above for teachers to see how to install R packages from a Dockerfile - although perhaps we should choose to install some package which is not pre-installed in the base image.

---

I will test the class script without installing the packages to confirm `tidyverse` is enough. I will then change the script to use other packages not included in the base image.

---

OK. 

If you prepare a script for `class_02`, can you use a package in that script that is not present in the base image and then install the package from the Dockerfile? Just to demonstrate how this is done for people who will be using this course for inspiration.